core/src/test/java/com/google/googlejavaformat/java/FormatterTest.java

```diff
@@ -418,4 +418,78 @@ public void throwsFormatterException() throws Exception {
     } catch (FormatterException expected) {
     }
   }
+
+  @Test
+  public void indentsIfStatementsWithoutBracesButWithCommentsOnTheSameLine()
+      throws FormatterException {
+    assertThat(
+            new Formatter()
+                .formatSource(
+                    "public class CommentedTest {\n"
+                        + "\n"
+                        + "  public boolean compare(ProglangType other) {\n"
+                        + "    if (this == other) // comment\n"
+                        + "    return true;\n"
+                        + "    return false;\n"
+                        + "  }\n"
+                        + "}\n"))
+        .isEqualTo(
+            "public class CommentedTest {\n"
+                + "\n"
+                + "  public boolean compare(ProglangType other) {\n"
+                + "    if (this == other) // comment\n"
+                + "     return true;\n"
+                + "    return false;\n"
+                + "  }\n"
+                + "}\n");
+  }
+
+  @Test
+  public void indentsDoWhileLoopsWithoutBracesButWithCommentsOnTheSameLine()
+      throws FormatterException {
+    assertThat(
+            new Formatter()
+                .formatSource(
+                    "public class CommentedTest {\n"
+                        + "\n"
+                        + "  public boolean something(ProglangType other) {\n"
+                        + "    do // comment\n"
+                        + "    return true;\n"
+                        + "    while (true);\n"
+                        + "  }\n"
+                        + "}\n"))
+        .isEqualTo(
+            "public class CommentedTest {\n"
+                + "\n"
+                + "  public boolean something(ProglangType other) {\n"
+                + "    do // comment\n"
+                + "     return true;\n"
```

Comment:
I also think that the line generated here by google-java-format should be indented +1.